src/rebar3_hex_organization.erl

```diff
@@ -213,20 +213,20 @@ format_error({get_repo_by_name, {error,{not_valid_repo,ParentName}}}) ->
 format_error(Reason) ->
     rebar3_hex_error:format_error(Reason).
 
--dialyzer({nowarn_function, auth/2}).
 -spec auth(rebar_state:t(), binary()) -> {ok, rebar_state:t()}.
 auth(State, RepoName) ->
     {Opts, _} = rebar_state:command_parsed_args(State),
-    {ParentRepo, OrgName} = get_parent_repo_and_org_name(State, RepoName),
+    {ParentRepo0, OrgName} = get_parent_repo_and_org_name(State, RepoName),
     Key =
         case proplists:get_value(key, Opts, undefined) of
             undefined ->
-                Config = rebar3_hex_config:get_hex_config(?MODULE, ParentRepo, write),
+                Config = rebar3_hex_config:get_hex_config(?MODULE, ParentRepo0, write),
                 Config1 = Config#{api_organization => OrgName},
                 KeyName = proplists:get_value(key_name, Opts, rebar3_hex_config:repos_key_name()),
                 generate_key(Config1, KeyName, default_perms(OrgName));
             ProvidedKey ->
                 TestPerms = #{domain => <<"repository">>, resource => OrgName},
+                ParentRepo = maps:without([name], ParentRepo0),
```

Comment:
The spec doesn't mention name, and I'm also not sure why it's there, but it is used internally, by `rebar3_hex` though. Making this call fixes the Dialyzer issue and gets us closer to the current spec.

Reply:
@starbelly, do you know where `name` comes from? Is it potentially just a quick local hack? Are you Ok with the current solution of removing it to obey the spec?

Reply:
`name` comes from repo config. Example : 

```
hex, [{doc, ex_doc},
        {repos, [
                #{
                    name => <<"hexpm:myrepo">>,
                    api_url => <<"http://localhost:4000/api">>,
                    repo_url => <<"http://localhost:4000">>,
                    repo_verify => false,
                    repo_verify_origin => false
                },
                #{
                    name => <<"hexpm">>,
                    api_url => <<"http://localhost:4000/api">>,
                    repo_url => <<"http://localhost:4000">>,
                    repo_verify => false,
                    repo_verify_origin => false
                }]}]}.
```

Reply:
Hm... so should it be a key in the map spec? Or why is the code working if I remove it (assuming it's tested, but probably it's not 😢)

Reply:
Well, it definitely works, I use it :) It is tested as part of the "integration" test suite as well. I suppose an interesting question, is why you expect it to fail? Perhaps you are thinking dialyzer should catch this?

Reply:
> Well, it definitely works, I use it :)

I meant "I removed it, and it seems to 1. stop Dialyzer complaining, 2. not stop tests" (which is what I was referring to with the coverage part of my comment).

Reply:
> why you expect it to fail?

If it's there (prior to this pull request) I'd expect it to fail if I removed it and it was required. Dialyzer complaining does not mean it's not working, it just means there's either a wrong spec or a wrong call...

Reply:
> Perhaps you are thinking dialyzer should catch this?

Dialyzer definitely catches that we're calling a function with a map containing a key named `name` when it's not expected, but that doesn't mean there's an error, so I'm leaning toward an incomplete spec.

Reply:
In summary: [in `hex_core`](https://github.com/hexpm/hex_core/blob/main/src/hex_core.erl#L90) the key appears to be named `repo_name` (not `name`), and it seems used [here](https://github.com/hexpm/hex_core/blob/main/src/hex_repo.erl#L197), but it's also possible that code is not used, and the name is always returned by the previous line of code.

Reply:
Since this started to feel familiar, and went to search the code then the Git history, and I seem to have [made a similar change in the past](https://github.com/erlef/rebar3_hex/pull/229/files#diff-f91a12f48327c7c0b04d92bc558861bb13dbdbcff61f9f72f45c47ed2e8f8c1aR119), that got accepted and is in tag-released code. Should I create a function that identifies the goal of removing the key and use it in the existing places and the new places I just tweaked?

Also, is it possible we're using that key internally (`rebar3_hex` only) and not in `hex_core` since 1. there's working code without it, 2. Dialyzer complains, 3. tests don't fail without it... (?)

Reply:
Ahh I see, apologies, I should have been more clear. It is indeed internal to rebar3 and rebar3_hex, hex_core doesn't care about that key. Since the call would succeed, at the time it was deemed pragmatic vs trying to remove all keys that are not in `hex_core:config/0` for the sake of doing so.

Reply:
Yeah, that's when I think by default Dialyzer should not care about extra keys in maps, but I can see benefits in both sides of the question.

Reply:
`hex_core` has `repo_name`. Do you think it's retro-fittable, in that case? Stop using `name` and using `repo_name`?

Reply:
I've updated the current code (it seems `name` goes into `rebar3` which makes it more difficult to just use expected `repo_name`) by concentrating it in a single module: [src/rebar3_hex_client.erl](https://github.com/erlef/rebar3_hex/pull/338/files#diff-d27dd061e87ebcd5b13f06bad04d32afd9e3351ccbc771abab65d9e63de828c0).

Reply:
I did a bit of digging, if for nothing more than a stroll down memory lane :) Prior to 2019 there was no `repo_name`, anywhere, `name` was the cool kid on the block! :p  hex_core added `repo_name` in https://github.com/hexpm/hex_core/pull/48 . Likewise, this [https://github.com/erlang/rebar3/pull/1996] adjusted rebar to be compat with hex_core 0.4.0 and finally the first mention of repo_name in rebar3_hex is [here](https://github.com/erlef/rebar3_hex/pull/105). 

I think this was to disambiguate the parent (repository) from the child (org). 

And yes, I don't we can just stop using name. It's a worthy discussion to be had outside the scope of this PR though!

Reply:
Got it. Lemme know if you prefer to ignore the Dialyzer, or keep the implementation I'm suggesting... (with the small wrapper function)